docs/usage/config-presets.md

```diff
@@ -232,6 +232,44 @@ Parameters are supported similar to other methods:
 }
 ```
 
+## Templating presets
+
+In order to provide flexible preset configurations, Renovate provides support using [templates](./templates.md) for preset names.
+
+### Example use-case
+
+The following example shows a self-hosted Renovate setup (`config.js`) within a Git repository `my-org/renovate` of a self-hosted GitLab instance.
+Additionally, there is a shareable config preset configured within the same repository (`default.json`).
+
+Simplified repository tree
+
+```
+- .gitlab-ci.yml
+- config.js
+- default.json
+```
+
+Renovate is configured to use the shareable preset as a base configuration.
+
+```javascript
+module.export = {
+  extends: ['local>my-org/renovate'],
+};
+```
+
+However, using this approach Renovate only loads the preset from the default branch of the repository (usually called `main`).
+In case you want to apply changes to the preset, e.g. via opening a new merge-request, the changes should be validated to not break the Renovate setup (e.g. using the `renovate-config-validator`).
+Here the templating can be used to create a dynamic reference, which can be used by Renovate to load the correct preset (else it would just load the preset from the default branch).
+
+```javascript
+module.exports = {
+  customEnvVariables: {
+    GITLAB_REF: process.env.CI_COMMIT_REF_NAME,
+  },
+  extends: ['local>my-org/renovate#{{ env.GITLAB_REF }}'],
+};
+```
```

Comment:
Is this your actual use case?

I'm not sure if this will confuse users more than help them. i.e. the 99% who don't need this start wondering if they'll miss something.

I thought your use case was to be able to make the org name relative

Reply:
Sure I can add a more simple use case

Reply:
Sth. like this? Using a preset based on organisation:
```json
{
  "extends": [
    "config:recommended",
    "local>{{ env.ORG_SLUG }}/renovate"
  ]
}
```

Reply:
your use case is that this used only with env variables?

Reply:
We can also use `topLevelOrg` for the use case mentioned above.